feat: ✨ added automatic extraction of .wabbajack files, GUI and GitHub action to compile into EXE and put out a release

DassaultMirage2K opened the pull request:
First: I added automatic extraction of `modlist` file inside of `.wabbajack` removing the requirement of extracting it manually.
```python
 with zipfile.ZipFile(filename, 'r') as zipObj:
                metadata_name = "modlist"
                with zipObj.open(metadata_name, 'r') as metadata:
                    extract_url(json.loads(metadata.read().decode('utf-8').replace("'", '"')))
```

Second: I added GUI using tkinter from Python standard library:
![app interface](https://github.com/user-attachments/assets/091fd787-89ba-48df-81d3-3883298f6348)
This added an option to select .wabbajack file from anywhere on the PC

.wabbajack file extract example:
![example 1](https://github.com/user-attachments/assets/66755a3b-b9a6-45a5-8d86-6a7d70a96552)


Download batch opens up web browser like before:
![example 2](https://github.com/user-attachments/assets/9d2ea582-ec61-4099-9461-89b45e669aad)

Third: Added Github action to automatically compile the app and publish on tag push with `v*.*.*` template.  
Example:
https://github.com/DassaultMirage2K/Wabbajack-fast-downloader/releases/tag/v1.0.5

DassaultMirage2K committed:
Added GUI with tkinter

DassaultMirage2K committed:
Added GitHub action

DassaultMirage2K committed:
Action test

DassaultMirage2K committed:
Include tk-inter plugin

DassaultMirage2K committed:
Disable console and fix Uplaod Release Asset

DassaultMirage2K committed:
Added automatic import of output file if its exits on start or after extraction

DassaultMirage2K committed:
Fixed tag name in Upload Release Asset exe name

DassaultMirage2K committed:
Fixed label instruction so its always up to date

DassaultMirage2K committed:
Fixed action order in console output

M1n-74316D65 commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,54 @@
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.12'
+          architecture: 'x64'
```
Please update to newer versions if possible.

M1n-74316D65 commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,54 @@
+          name: windows-latest Build
+          path: | 
+            build/*.exe
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref_name }}
+          release_name: ${{ github.ref_name }}
+          body: |
+            ${{ github.event.head_commit.message }}
+          draft: false
+          prerelease: false
```
create-release has been deprecated since 2021, please fix

M1n-74316D65 commented on gui.py:
It looks good. Thank you.

M1n-74316D65 commented:
On Wednesday, I can do it if you don't feel like changing it. Thanks 😃

DassaultMirage2K committed:
Updated the readme with new instructions. Updated job actions to newer versions.

DassaultMirage2K committed:
Changed Create Release to softprops/action-gh-release

DassaultMirage2K committed:
Fixed filepath for Create Release

DassaultMirage2K committed:
name fix

DassaultMirage2K commented:
Added executable usage to README, just need to change the link to release page. 
Updated actions: Nuitka compiler doesn't support Python 3.13 for now.

M1n-74316D65 committed:
docs(README): :memo: change executable usage place and executable URL

M1n-74316D65 commented:
Thanks
